includes/Handlers/Product/Woo_SOR.php

```diff
@@ -225,7 +290,102 @@ public static function update_catalog_variation( CatalogObject $catalog_object,
 		 * @see https://github.com/woocommerce/woocommerce-square/issues/570
 		 */
 		if ( 'variation' === $product->get_type() ) {
-			$variation_data->setName( $product->get_name() );
+			$options_data_transient = get_transient( 'wc_square_options_data' );
+			$parent_product         = wc_get_product( $product->get_parent_id() );
+			$attributes             = $parent_product->get_attributes();
+			$variation_items        = $product->get_attributes();
+			$variation_item_values  = array();
+
+			if ( 1 === count( $attributes ) ) {
+				// Set the name of the variation if it's a single variation.
+				$variation_data->setName( reset( $variation_items ) );
+				$variation_data->setItemOptionValues( null );
+			} else {
+				// If there are multiple attributes, the name of the variation is the combination of all attribute values.
+				$variation_name = array();
+
+				/**
+				 * Set the `item_option_values` for the variation.
+				 *
+				 * Retrieve the options data from the transient. At this point, the options data
+				 * should already be available, as we have already created the necessary options
+				 * and values in the parent product above.
+				 */
+				foreach ( $variation_items as $attribute_id => $attribute_value ) {
+					// Check if it's a global attribute (taxonomy-based, e.g., "pa_color")
+					$taxonomy_exists = false;
+					if ( taxonomy_exists( $attribute_id ) ) {
+						// Use wc_attribute_label for global attributes
+						$attribute_name   = $attribute_id;
+						$variation_name[] = $attribute_value = get_term_by( 'slug', $attribute_value, $attribute_id )->name;
+						$taxonomy_exists  = true;
+					} else {
+						// For custom attributes, simply use the cleaned-up attribute ID
+						$attribute_name   = ucwords( str_replace( '-', ' ', $attribute_id ) ); // what if merchant changes the name?
```

Comment:
Maybe change this to a `TODO` comment to make it easier to find in the future? Or remove all together if we have this tracked elsewhere